var/spack/repos/builtin/packages/hipsparse/package.py

```diff
@@ -39,6 +39,7 @@ class Hipsparse(CMakePackage):
 
     patch('e79985dccde22d826aceb3badfc643a3227979d2.patch', when='@3.5.0')
     patch('530047af4a0f437dafc02f76b3a17e3b1536c7ec.patch', when='@3.5.0')
+    patch('hipsparse-remove-opt-rocm-reference.patch', when='@4.2.0:')
```

Comment:
Can you share what system version got picked up?

Also here, it should pick up spack's hip.

Reply:
i have attached the CMakeCache.txt with out the patch, i had a rocm-4.3.0 installed in the /opt/rocm path and when i build the hipblas and hipsparse recipes using spack, the build succeeds but the cmakecache.txt seems to refer to few files from /opt/rocm  for HIP_run_hipcc etc.